File: FreelanceMarketplace/FreelanceMarketplace/GraphQL/Types/ProjectType.cs
using FreelanceMarketplace.Models;
using GraphQL.Types;
using System.Management;

namespace FreelanceMarketplace.GraphQL.Types
{
    public class ProjectType : ObjectGraphType<Project>
    {
        public ProjectType()
        {
            Field(x => x.ProjectId);
            Field(x => x.ProjectName);
            Field(x => x.ProjectDescription);
            Field(x => x.Budget);
            Field(x => x.Deadline);
            Field(x => x.SkillRequire);
            Field(x => x.Status);
            Field(x => x.CreateAt);

            Field<CategoryType>("category");
            Field<UserType>("user").Resolve(context => context.Source.Users);
            Field<ListGraphType<ApplyType>>("applies");
            Field<ListGraphType<StringGraphType>>("imageUrls")
            .Resolve(context => context.Source.Images.Select(i => i.ImageUrl));
        }
    }
}

File: FE/src/types/ProjectType.ts
import { ApplyType } from "./ApplyType";
import { CategoryType } from "./CategoryType";
import { Contract } from "./ContractType";
import { User } from "./UserType";

export interface ProjectResponseType<T> {
    data: T;
}

export interface ProjectType {
    projectId: number;
    projectName: string;
    projectDescription: string;
    budget: number;
    deadline: string;
    skillRequire: string;
    status: "active" | "completed" | "processing";
    createAt: string;
    users?: User;
    category?: CategoryType;
    applies: ApplyType[];
    contract: Contract;
}

export interface ProjectInput {
    projectName: string;
    projectDescription: string;
    budget: number;
    deadline: string;
    skillRequired: string;
    status: string;
    categoryId: string;
}


export interface ProjectImageResponse {
    data: {
        projects: ProjectWithImage[]
    }
}

export interface ProjectWithImage {
    projectId: number;
    projectName: string;
    projectDescription: string;
    budget: number;
    deadline: Date;
    skillRequire: string;
    status: string;
    createAt: Date;
    avatar: string;
    category: CategoryType;
    imageUrls: string[];
    users: User;
}